### PR TITLE
[FIX] kmer_hash combined with reverse

### LINKED ISSUE #2415: [BUG] kmer_hash and reverse for a list of one results in wrong result
```cpp
    std::vector<seqan3::dna4> text1{"TAAA"_dna4};
    std::vector<seqan3::dna4> text2{"TTTA"_dna4};
    auto v =  seqan3::views::kmer_hash(text1, seqan3::ungapped{4}) | std::views::reverse;
    auto v2 = text2 | seqan3::views::kmer_hash(seqan3::ungapped{4}) | std::views::reverse;
    seqan3::debug_stream << v << " " << v2<< "\n";
```

Results in: [48] [63]
It should be: [192] [252]

48 would be true for TAA and 63 for TTT. I don't know why the reverse leads to a wrong kmer_hash result. If I remove the std::views::reverse it is working as expected.

### PR DESCRIPTION
**Pull request by eseiler**
Resolves #2415

### CODE REVIEW

File: include/seqan3/range/views/kmer_hash.hpp [hunk 1]
```diff
@@ -327,16 +327,15 @@ class kmer_hash_view<urng_t>::basic_iterator
         assert(std::ranges::size(shape_) > 0);
 
         auto urange_size = std::ranges::distance(it_start, it_end);
-        auto step = (shape_.size() + 1 > urange_size) ? 0 : urange_size - shape_.size() + 1;
+        auto step = (shape_.size() > urange_size + 1) ? 0 : urange_size - shape_.size() + 1;
```
Comment (eseiler) [inline include/seqan3/range/views/kmer_hash.hpp:330]: This was the bug

File: include/seqan3/range/views/kmer_hash.hpp [hunk 2]
```diff
@@ -327,16 +327,15 @@ class kmer_hash_view<urng_t>::basic_iterator
...
         text_left = std::ranges::next(it_start, step, it_end);
-        text_right = it_end;
```
Comment (eseiler) [inline include/seqan3/range/views/kmer_hash.hpp:332]: This was an unnecessary assignment

**Review by MitraDarja — APPROVED** [review]
Thanks for fixing this so fast. 😍

File: include/seqan3/range/views/kmer_hash.hpp (hunk 2 above)
Comment (marehr) [reply include/seqan3/range/views/kmer_hash.hpp:332]: Where is text_right defined? Can it be removed?

**Review by marehr — COMMENTED** [review]
.

File: include/seqan3/range/views/kmer_hash.hpp (hunk 2 above)
Comment (eseiler) [reply include/seqan3/range/views/kmer_hash.hpp:332]: It's a class member, cannot be removed. See `\details` of this constructor as to why we have `it_end`.

But what happened here was:
* `text_right = it_end` at this line
* Maybe calling `hash_full` which immediately does `text_right = text_left`
* `text_right = it_end` at the end

**Review by marehr — APPROVED** [review]

**Pull request merged**